hail/src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -786,6 +792,37 @@ abstract class PartitionWriter {
   def toJValue: JValue = Extraction.decompose(this)(PartitionWriter.formats)
 }
 
+abstract class SimplePartitionWriter extends PartitionWriter {
+  def ctxType: Type = TString
+  def returnType: Type = TString
+  def unionTypeRequiredness(r: TypeWithRequiredness, ctxType: TypeWithRequiredness, streamType: RIterable): Unit = {
+    r.union(ctxType.required)
+    r.union(streamType.required)
```

Comment:
I think you're supposed to `unionFrom` here? The `ctxType` itself could be a struct with fields that are potentially missing.

Reply:
I don't really understand this part of the code. I basically just copied it from the PartitionNativeWriter.